src/App.js

```diff
@@ -1,16 +1,37 @@
 import "./App.css";
 import React, { useState } from "react";
 import Auth from "./pages/Auth";
-import { AuthContext } from "./common/contexts";
 import Home from "./pages/Home";
+import Signup from "./pages/Signup";
+import Login from "./pages/Login";
+import { UserContext, CredContext } from "./common/contexts";
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import PrivateRoute from "./common/privateRoute";
 
 function App() {
-  const [auth, setAuth] = useState(false);
+  const [user, setUser] = useState({ email: "", password: "", auth: false });
+  const [creds, setCreds] = useState([]);
 
   return (
-    <AuthContext.Provider value={{ auth, setAuth }}>
-      <div className="App">{auth ? <Home /> : <Auth />}</div>
-    </AuthContext.Provider>
+    <UserContext.Provider value={{ user, setUser }}>
```

Comment:
Weak structure. We should use page based or module based contexts. Not at the application level

Reply:
Now, I am using the provider component but still wrapping the whole routes inside it. The other option that comes to my mind is to render each component wrapped in providerContext on each route, but I think so that would require some repeatable code.